docs/README.Android.md

```diff
@@ -0,0 +1,305 @@
+```
+
+**TIP:** By adding `-j<number>` to the make command, you can choose how many concurrent jobs will be used and expedite the build process. It is recommended to use `-j$(getconf _NPROCESSORS_ONLN)` to compile on all available processor cores. The build machine can also be configured to do this automatically by adding `export MAKEFLAGS="-j(getconf _NPROCESSORS_ONLN)"` to your shell config (e.g. `~/.bashrc`).
+
+**WARNING:** Look for the `Dependencies built successfully.` success message. If in doubt run a single threaded `make` command until the message appears. If the single make fails, clean the specific library by issuing `make -C target/<name_of_failed_lib> distclean` and run `make`again.
+
+
+**[back to top](#table-of-contents)** | **[back to section top](#5-build-tools-and-dependencies)**
+
+## 6. Build binary add-ons
+You can find a complete list of available binary add-ons **[here](https://github.com/xbmc/repo-binary-addons)**.
+
+Change to Kodi's source code directory:
+```
+cd $HOME/kodi
+```
+
+Build all add-ons:
+```
+make -j$(getconf _NPROCESSORS_ONLN) -C tools/depends/target/binary-addons
+```
+
+Build specific add-ons:
+```
+make -j$(getconf _NPROCESSORS_ONLN) -C tools/depends/target/binary-addons ADDONS="audioencoder.flac pvr.vdr.vnsi audiodecoder.snesapu"
+```
+
+Build a specific group of add-ons:
+```
+make -j$(getconf _NPROCESSORS_ONLN) -C tools/depends/target/binary-addons ADDONS="pvr.*"
+```
+
+**[back to top](#table-of-contents)**
+
+## 7. Build Kodi
+Configure CMake build:
+```
+cd $HOME/kodi
+make -C tools/depends/target/cmakebuildsys
+```
+
+Build Kodi:
+```
+cd $HOME/kodi/build
+make -j$(getconf _NPROCESSORS_ONLN)
+```
+
+**[back to top](#table-of-contents)**
+
+## 8. Package
+CMake generates a target called `apk` which will package Kodi ready for distribution.
+
+Create package:
+```
+make apk
+```
+
+Generated `apk` file will be inside `$HOME/kodi`.
+
+**[back to top](#table-of-contents)**
+
+## 9. Install
+Connect your Android device to your computer through USB and enable the `Unknown sources` option in your device settings.
+
+Make sure `adb` is installed:
+```
+sudo apt install adb
+```
+
+Install Kodi:
+```
+cd $HOME/kodi-android
+adb devices
+adb -s <device-id> install -r <generated-apk-name-here>.apk
+```
+
+The *device-id* can be retrieved from the list returned by the `adb devices` command and is the first value in the row representing your device.
+
+**[back to top](#table-of-contents)**
+
+## 10. Debugging Kodi
+To be able to see what is happening while running Kodi you need to enable `USB debugging` in your device settings (already enabled in the Android Emulator).
+
+Access the log output of your Android device:
+```
+adb -s <device-id> logcat
+```
+
+Install a new build over the existing one:
+```
+adb -e install -r images/xbmcapp-debug.apk
+```
+
+Launch Kodi on Android Emulator without the GUI:
+```
+adb shell am start -a android.intent.action.MAIN -n org.xbmc.xbmc/android.app.NativeActivity
```

Comment:
Found an error: `org.xbmc.kodi` -- though the command still doesn't work for me, I get:

```
Starting: Intent { act=android.intent.action.MAIN cmp=org.xbmc.kodi/android.app.NativeActivity }
Error type 3
Error: Activity class {org.xbmc.kodi/android.app.NativeActivity} does not exist.
```